doc/tutorials/12-constant_pH/12-constant_pH.ipynb

```diff
@@ -166,130 +314,251 @@
     "\n",
-    "\\begin{equation}\n",
-    "    \\mathrm{CI}_{95\\%} = \\bar{\\alpha} \\pm z_{1-0.95/2} \\cdot \\frac{s_{n-1}}{\\sqrt{n}}\n",
-    "\\end{equation}\n",
+    "To achieve a basic pH we need to add even more base to the system to neutralize the polymer.\n",
+    "Again, there are 4 ionic species in the solution:  $\\mathrm{H^+}$, $\\mathrm{A^-}$, $\\mathrm{OH^-}$, and $\\mathrm{M^+}$ and we can not clearly attribute a specific chemical identity to the $\\mathrm{B^+}$ ions. Because only very few $\\mathrm{H^+}$ ions should be present in the solution, we can make the approximation that at this pH, all $\\mathrm{A^-}$ ions are neutralized by the $\\mathrm{M^+}$ ions, and therefore $\\mathrm{B^+}$ ions in the simulation correspond to $\\mathrm{M^+}$ ions in the real solution. Because additional $\\mathrm{M^+}$ ions in the real solution neutralize the $\\mathrm{OH^-}$ ions, the concentration of $\\mathrm{B^+}$ does not correspond to the concentration of $\\mathrm{M^+}$ ions. Now, again many ions are missing in the simulation box, as compared to the real solution: Few $\\mathrm{H^+}$ ions, many $\\mathrm{OH^-}$ ions, and a comparable amount of the $\\mathrm{M^+}$ ions.\n",
     "\n",
-    "We will use that estimate to plot the error bars."
+    "To further illustrate this subject, we compare the concentration of the neutralizer ion $\\mathrm{B^+}$ calculated in the simulation with the expected number of ions of each species. At a given pH and pK we can calculate the expected degree of ionization from the Henderson Hasselbalch equation. Then we apply the electroneutrality condition \n",
+    "$$c_\\mathrm{A^-} + c_\\mathrm{OH^-} + c_\\mathrm{X^-} = c_\\mathrm{H^+} + c_\\mathrm{M^+}$$\n",
+    "where we use either $c_\\mathrm{X^-}=0$ or $c_\\mathrm{M^+}=0$ because we always only add extra acid or base, but never both. Adding both would be equivalent to adding extra salt $\\mathrm{MX}$.\n",
+    "We obtain the concentrations of $\\mathrm{OH^-}$ and $\\mathrm{H^+}$ from the input pH value, and substitute them to the electroneutrality equation to obtain\n",
+    "$$\\alpha c_\\mathrm{acid} + 10^{-(\\mathrm{p}K_\\mathrm{w} - \\mathrm{pH})} + 10^{-\\mathrm{pH}} = c_\\mathrm{M^+} - c_\\mathrm{X^-}$$\n",
+    "Depending on whether the left-hand side of this equation is positive or negative we know whether we should add $\\mathrm{M^+}$ or $\\mathrm{X^-}$ ions."
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {},
+   "metadata": {
+    "scrolled": false
+   },
    "outputs": [],
    "source": [
-    "import scipy.stats\n",
-    "def ideal_degree_of_dissociation(pH, pK):\n",
-    "    return 1. / (1 + 10**(pK - pH))\n",
+    "av_c = av_alpha*c_acid # average concentration of B+ in mol/L\n",
+    "err_c = err_alpha*c_acid # error in the average concentration\n",
```

Comment:
Please drop the comments they easily get outdated.

Reply:
@jonaslandsgesell I do not understand. What could become outdated in these comments?

Reply:
the information that av_c is in mol/l is redundant, because this is carried over from c_acid. This can become outdated if someone changes the unit of c_acid.
You might consider variable names which are more self-explaining. Then the comment is also not needed anymore.

Reply:
I will make the variable names more explicit.